tsdb/agent/db_test.go

```diff
@@ -68,93 +113,74 @@ func TestCommit(t *testing.T) {
+
+			e := exemplar.Exemplar{
+				Labels: lset,
+				Ts:     sample[0].T(),
+				Value:  sample[0].V(),
+				HasTs:  true,
+			}
+			_, err = app.AppendExemplar(ref, lset, e)
 			require.NoError(t, err)
 		}
 	}
 
-	require.NoError(t, a.Commit())
+	require.NoError(t, app.Commit())
 	require.NoError(t, s.Close())
 
-	// Read records from WAL and check for expected count of series and samples.
-	walSeriesCount := 0
-	walSamplesCount := 0
-
-	reg = prometheus.NewRegistry()
-	remoteStorage = remote.NewStorage(log.With(logger, "component", "remote"), reg, startTime, promAgentDir, time.Second*30, nil)
-	defer func() {
-		require.NoError(t, remoteStorage.Close())
-	}()
-
-	s1, err := Open(logger, nil, remoteStorage, promAgentDir, opts)
+	sr, err := wal.NewSegmentsReader(s.wal.Dir())
 	require.NoError(t, err)
 	defer func() {
-		require.NoError(t, s1.Close())
+		require.NoError(t, sr.Close())
 	}()
 
-	var dec record.Decoder
+	// Read records from WAL and check for expected count of series and samples.
+	var (
+		r   = wal.NewReader(sr)
+		dec record.Decoder
+
+		walSeriesCount, walSamplesCount, walExemplarsCount int
+	)
+	for r.Next() {
+		rec := r.Record()
+		switch dec.Type(rec) {
+		case record.Series:
+			var series []record.RefSeries
+			series, err = dec.Series(rec, series)
+			require.NoError(t, err)
+			walSeriesCount += len(series)
 
-	if err == nil {
-		sr, err := wal.NewSegmentsReader(s1.wal.Dir())
-		require.NoError(t, err)
-		defer func() {
-			require.NoError(t, sr.Close())
-		}()
-
-		r := wal.NewReader(sr)
-		seriesPool := sync.Pool{
-			New: func() interface{} {
-				return []record.RefSeries{}
-			},
-		}
-		samplesPool := sync.Pool{
-			New: func() interface{} {
-				return []record.RefSample{}
-			},
-		}
+		case record.Samples:
+			var samples []record.RefSample
+			samples, err = dec.Samples(rec, samples)
+			require.NoError(t, err)
+			walSamplesCount += len(samples)
 
-		for r.Next() {
-			rec := r.Record()
-			switch dec.Type(rec) {
-			case record.Series:
-				series := seriesPool.Get().([]record.RefSeries)[:0]
-				series, _ = dec.Series(rec, series)
-				walSeriesCount += len(series)
-			case record.Samples:
-				samples := samplesPool.Get().([]record.RefSample)[:0]
-				samples, _ = dec.Samples(rec, samples)
-				walSamplesCount += len(samples)
-			default:
-			}
+		case record.Exemplars:
+			var exemplars []record.RefExemplar
+			exemplars, err = dec.Exemplars(rec, exemplars)
+			require.NoError(t, err)
+			walExemplarsCount += len(exemplars)
+
+		default:
 		}
 	}
 
-	// Retrieved series count from WAL should match the count of series been added to the WAL.
-	require.Equal(t, walSeriesCount, numSeries)
-
-	// Retrieved samples count from WAL should match the count of samples been added to the WAL.
-	require.Equal(t, walSamplesCount, numSeries*numDatapoints)
+	// Check that the WAL contained the same number of commited series/samples/exemplars.
+	require.Equal(t, walSeriesCount, numSeries, "unexpected number of series")
```

Comment:
The expected and actual seems to be swapped here (and below). Expected is the 2nd argument while actual should be the 3rd argument.